On this:

> Removes -Wdeclaration-after-statement. See the following message within this PR for the context.

It's either this, or fixing the existing issues (many I suspect, but some come from common headers):

<details>
<summary> Some errors... </summary>

```
bpf_ct_tests.c:42:16: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
   42 |         struct iphdr *l3 = dst;
      |                       ^
bpf_ct_tests.c:95:9: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
   95 |                 void *data = (void *)(long)ctx->data;
      |                       ^
bpf_ct_tests.c:140:20: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
  140 |                 struct ct_entry *entry = map_lookup_elem(get_ct_map4(&tuple), &tuple);
      |                                  ^
bpf_ct_tests.c:104:2: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
  104 |         TEST("ct4_syn", {
      |         ^
./common.h:233:20: note: expanded from macro 'TEST'
  233 |         static const char ____name[] = name;                               \
      |                           ^
bpf_ct_tests.c:200:20: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
  200 |                 struct ct_entry *entry = map_lookup_elem(get_ct_map4(&tuple), &tuple);
      |                                  ^
bpf_ct_tests.c:165:2: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
  165 |         TEST("ct4_rst", {
      |         ^
./common.h:233:20: note: expanded from macro 'TEST'
  233 |         static const char ____name[] = name;                               \
      |                           ^
bpf_ct_tests.c:87:6: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
   87 |         int pkt_size = mkpkt(pkt, true);
      |             ^
bpf_ct_tests.c:85:2: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
   85 |         test_init();
      |         ^
./common.h:219:23: note: expanded from macro 'test_init'
  219 |         __maybe_unused char *test_result_cursor = 0;                      \
      |                              ^
8 errors generated.
```
</details>

cc @ti-mo

--- a/bpf/tests/Makefile
+++ b/bpf/tests/Makefile
@@ -7,19 +7,15 @@ MAKEFLAGS += -r
 
 CLANG ?= clang
 
-FLAGS := -I$(ROOT_DIR)/bpf -I$(ROOT_DIR)/bpf/include -O2 -g
+FLAGS := -I$(ROOT_DIR)/bpf -I$(ROOT_DIR)/bpf/include -g
 
-CLANG_FLAGS := ${FLAGS} --target=bpf -std=gnu99 -nostdinc
-# eBPF verifier enforces unaligned access checks where necessary, so don't
-# let clang complain too early.
-CLANG_FLAGS += -Wall -Wextra -Werror -Wshadow
-CLANG_FLAGS += -Wno-address-of-packed-member
-CLANG_FLAGS += -Wno-unknown-warning-option
-CLANG_FLAGS += -Wno-gnu-variable-sized-type-not-at-end
-CLANG_FLAGS += -Wimplicit-int-conversion -Wenum-conversion
-CLANG_FLAGS += -Wimplicit-fallthrough
+# Base CFLAGS (CLANG_FLAGS) are read from pkg/datapath/loader/compile.go
+# to keep them synced with the loader
+CLANG_FLAGS := $(FLAGS) $(shell $(GO) run $(ROOT_DIR)/pkg/datapath/loader/tools/clang_cflags.go)
+
 # Create dependency files for each .o file.
 CLANG_FLAGS += -MD
+
 # Mimics the mcpu values set by cilium-agent. See GetBPFCPU().
 ifneq ($(KERNEL),54)
 CLANG_FLAGS += -mcpu=v3

--- a/pkg/datapath/loader/compile.go
+++ b/pkg/datapath/loader/compile.go
@@ -97,7 +97,7 @@ type directoryInfo struct {
 }
 
 var (
-	standardCFlags = []string{"-O2", "--target=bpf", "-std=gnu89",
+	standardCFlags = []string{"-O2", "--target=bpf", "-std=gnu99",
 		"-nostdinc",
 		"-Wall", "-Wextra", "-Werror", "-Wshadow",
 		"-Wno-address-of-packed-member",

--- a/pkg/datapath/loader/compile.go
+++ b/pkg/datapath/loader/compile.go
@@ -103,7 +103,6 @@ var (
 		"-Wno-address-of-packed-member",
 		"-Wno-unknown-warning-option",
 		"-Wno-gnu-variable-sized-type-not-at-end",
-		"-Wdeclaration-after-statement",
 		"-Wimplicit-int-conversion",
 		"-Wenum-conversion",
 		"-Wimplicit-fallthrough"}

--- a/pkg/datapath/loader/compile.go
+++ b/pkg/datapath/loader/compile.go
@@ -97,7 +97,7 @@ type directoryInfo struct {
 }
 
 var (
-	standardCFlags = []string{"-O2", "--target=bpf", "-std=gnu99",
+	StandardCFlags = []string{"-O2", "--target=bpf", "-std=gnu99",
 		"-nostdinc",
 		"-Wall", "-Wextra", "-Werror", "-Wshadow",
 		"-Wno-address-of-packed-member",
@@ -172,7 +172,7 @@ func compile(ctx context.Context, logger *slog.Logger, prog *progInfo, dir *dire
 		compileArgs = append(compileArgs, "-g")
 	}
 
-	compileArgs = append(compileArgs, standardCFlags...)
+	compileArgs = append(compileArgs, StandardCFlags...)
 	compileArgs = append(compileArgs, "-mcpu="+getBPFCPU(logger))
 	compileArgs = append(compileArgs, prog.Options...)
 	compileArgs = append(compileArgs,

--- a/pkg/datapath/loader/tools/clang_cflags.go
+++ b/pkg/datapath/loader/tools/clang_cflags.go
@@ -1,0 +1,15 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/datapath/loader"
+)
+
+func main() {
+	fmt.Println(strings.Join(loader.StandardCFlags, " "))
+}

